Model/Api/UpdateDiscountTrait.php

```diff
@@ -402,7 +402,7 @@ protected function applyingCouponCode($couponCode, $coupon, $quote, $addQuote =
             $quote->getBillingAddress() :
             $quote->getShippingAddress();
 
-        $discountAmount = ($address->getShippingMethod() && $this->cartHelper->ignoreAdjustingShippingAmount($quote))
+        $discountAmount = ($address->getShippingMethod() && $address->getShippingAmount() && $this->cartHelper->ignoreAdjustingShippingAmount($quote))
```

Comment:
Makes sense to me, but can we also  refactor this to IF ... ELSE and directly check if `$address->getCartFixedRules()[$ruleId]` set and bugsnag otherwise.